packages/cli/src/controller/migrate/manifest/ethereum.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {EthereumDatasourceKind, EthereumHandlerKind} from '@subql/common-ethereum';
+import {FileReference} from '@subql/types-core';
+import {
+  RuntimeDatasourceTemplate as EthTemplate,
+  SubqlBlockHandler as EthereumBlockHandler,
+  SubqlCallHandler as EthereumCallHandler,
+  SubqlEventHandler as EthereumEventHandler,
+  SubqlRuntimeDatasource as EthereumDs,
+} from '@subql/types-ethereum/dist/project';
+import {DEFAULT_HANDLER_BUILD_PATH} from '../../generate-controller';
+import {MigrateDatasourceKind, MigrateMappingType, SubgraphDataSource, SubgraphTemplate} from '../types';
+
+function baseDsConversion(ds: SubgraphDataSource | SubgraphTemplate): EthereumDs | EthTemplate {
+  const assets: Map<string, FileReference> = new Map();
+  for (const abi of ds.mapping.abis) {
+    assets.set(abi.name, {file: abi.file});
+  }
+
+  const subqlDs: EthereumDs | EthTemplate = {
+    kind: EthereumDatasourceKind.Runtime,
+    assets: new Map(ds.mapping.abis.map((a) => [a.name, {file: a.file}])),
+    mapping: {
+      file: DEFAULT_HANDLER_BUILD_PATH,
+      handlers: [
+        ...((ds.mapping.blockHandlers
+          ? ds.mapping.blockHandlers.map((h) => {
+              return {
+                kind: EthereumHandlerKind.Block,
+                handler: h.handler,
+              };
+            })
+          : []) as EthereumBlockHandler[]),
```

Comment:
I think this is more readable
```suggestion
        ...(ds.mapping.blockHandlers ?? []).map((h) => {
              return {
                kind: EthereumHandlerKind.Block,
                handler: h.handler,
              };
            })) as EthereumBlockHandler[]),
```